goal_src/jak1/pc/pckernel.gc

```diff
@@ -98,6 +98,9 @@
 
 (defmethod set-frame-rate! pc-settings ((obj pc-settings) (rate int))
   "set the target framerate."
+  ;; ignore changes to frame-rate while in speedrunner mode
+  (when (= (-> obj speedrunner-mode?) #t)
+        (set! rate 60))
```

Comment:
possibly redundant? You already set it to 60 when starting a run. Since you need to go to the menu to change fps anyway, rejecting the fps change unless you go to a *different* menu first seems a bit awkward.

Reply:
Possibly, i figured I would do this because this was the only setting exposed in the UI. But you are right, it would be visible in the recording so i can probably keep it simple

Reply:
Would it be better to just freeze this value in the function in autosplit.gc that keeps track of orbs/flies etc constantly?

It seems odd to allow a user to change a setting that would invalidate their run if we can just not?

This was it would all be separate from main code and only run when speedrun mode is on.

Reply:
There are many things that a user can do to invalidate their run in the menu, such as loading a save file.

Reply:
I think the nice (from both a user and dev's perspective) approach will be adding support to the progress menu code to allow disabling / hiding options -- which is a pretty basic GUI programming feature that would likely be nice to have regardless in the long-term.

Freezing the value -- aka writing it constantly -- has perf implications (writing to the settings file every frame), but the current approach of ignoring the value is pretty hacky as well.

I think for now it's for the best to keep it simple and be a part of verification until we solve it in a better way -- as dass says, there are plenty of ways someone can invalidate the run and a verifier should atleast scrub through to catch obvious things like this.

Reply:
Yeah it's hacky not ideal and confusing, but doesn't the settings file only update when you close the progress menu or autosave?

Reply:
Menu option is now nicely disabled when speedrunner mode is active, this feature could be taken advantage of for any menu option as well.